### PR TITLE
fix(display): keep max scale independent of rotation

### PR DESCRIPTION
**Pull request by Ivy233**
Calculate the scale limit with the long and short screen edges so rotated outputs use the same limit as the original display mode.

使用屏幕长短边计算缩放上限，使旋转后的输出使用与原始显示模式一致的上限。

Log: 修复旋转外接屏登录后缩放被重置的问题
Influence: 旋转外接屏后注销或重启，用户设置的缩放比例不会被错误重置。
PMS: BUG-365211

## Summary by Sourcery

Ensure maximum display scale factor remains consistent regardless of screen rotation by basing limits on the long and short edges of the display.

Bug Fixes:
- Prevent external rotated displays from having their user-configured scale factor incorrectly reduced and reset after login or reboot.

Tests:
- Add unit tests verifying calcMaxScaleFactor returns the same maximum scale for a display before and after 90°/180° rotation across multiple resolutions.

### CODE REVIEW

File: display1/util_test.go [hunk 1]
```diff
@@ -242,3 +242,28 @@ func Test_parseEdid(t *testing.T) {
 	}
 
 }
+
+// Test_calcMaxScaleFactor 校验最大可支持缩放与屏幕方向无关：90°/270° 旋转会交换
+// CRTC 宽高，旋转前后应得到相同的上限，避免因旋转把上限错误压低（如 1080×1920
+// 被压到 1.0）而在登录应用配置时触发 tryToChangeScaleFactor 重置用户缩放。
+func Test_calcMaxScaleFactor(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Ivy233) [reply display1/util_test.go:249]: 已添加。

**Review by fly602 — APPROVED** [review]
